pkg/kubelet/kubelet_pods.go

```diff
@@ -76,8 +80,71 @@ const (
 const (
 	PodInitializing   = "PodInitializing"
 	ContainerCreating = "ContainerCreating"
+
+	kubeletUser = "kubelet"
 )
 
+func parseGetSubIdsOutput(input string) (uint32, uint32, error) {
+	lines := strings.Split(strings.Trim(input, "\n"), "\n")
+	if len(lines) != 1 {
+		return 0, 0, fmt.Errorf("error parsing line %q: it must contain only one line", input)
+	}
+
+	parts := strings.Fields(lines[0])
+	if len(parts) != 4 {
+		return 0, 0, fmt.Errorf("invalid line %q", input)
+	}
+
+	// Parsing the numbers
+	num1, err := strconv.ParseUint(parts[2], 10, 64)
+	if err != nil {
+		return 0, 0, fmt.Errorf("error parsing line %q: %w", input, err)
+	}
+
+	num2, err := strconv.ParseUint(parts[3], 10, 64)
+	if err != nil {
+		return 0, 0, fmt.Errorf("error parsing line %q: %w", input, err)
+	}
+
+	return uint32(num1), uint32(num2), nil
+}
+
+// Get the additional IDs allocated for the Kubelet.
+func (kl *Kubelet) getKubeletMappings() (uint32, uint32, error) {
```

Comment:
We need to return the default mapping if the feature gate is disabled. Otherwise this can happen:

1. Wrong configuration of the kubelet user is done (i.e. not multiple of userns length)
2. The feature gate is disabled
3. Kubelet initialization still fails

Reply:
fixed